### PR TITLE
Feat/12 auth

### PR DESCRIPTION
**Pull request by FearsomeRover**
Added basic auth with the new awesome kir-dev package!

### CODE REVIEW

File: apps/backend/src/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private jwtService: JwtService
+  ) {}
+
+  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
+    if (user) return user;
+    return await this.prisma.user.create({
+      data: {
+        authSchId: userProfile.authSchId,
+        fullName: `${userProfile.lastName} ${userProfile.firstName}`, //this is kinda cring...
```
Comment (FearsomeRover) [inline apps/backend/src/auth/auth.service.ts:20]: Im not sure, how this handles third names here...

File: apps/backend/src/auth/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        nickName: userProfile.firstName,
+        email: userProfile.email,
+        isSchResident: true,
+        //TODO: Find a solution for only dorm residents. The userProfile.bmeStatus is undefined. Is it an admin enabled info?
```
Comment (FearsomeRover) [inline apps/backend/src/auth/auth.service.ts:24]: And this seems to not work here. Until we find a solution this seems useful for development purposes

**Comment by FearsomeRover** [comment]
Ohh, and I added Swagger, as we will hopefully use it later.

File: apps/backend/src/auth/auth.service.ts (hunk 1 above)
Comment (Tschonti) [reply apps/backend/src/auth/auth.service.ts:20]: the displayName scope and property from authSch is actually the full name. I thought it was the authsch username, but no. Might be worth renaming it in the package

File: apps/backend/src/auth/auth.service.ts (hunk 2 above)
Comment (Tschonti) [reply apps/backend/src/auth/auth.service.ts:24]: you need to add the BME_STATUS scope in the constructor to access that field. Still, that won't help with dorm residency.

File: apps/backend/src/main.ts [hunk 3]
```diff
@@ -1,9 +1,18 @@
 import { NestFactory } from '@nestjs/core';
+import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
 
 import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
-  await app.listen(3001);
+  app.enableCors({
+    origin: [process.env.FRONTEND_URL, 'http://localhost:3000'],
```
Comment (Tschonti) [inline apps/backend/src/main.ts:9]: maybe only put localhost here if we're not in production?

File: apps/backend/src/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
+    if (user) return user;
+    return await this.prisma.user.create({
+      data: {
+        authSchId: userProfile.authSchId,
+        fullName: userProfile.displayName,
+        nickName: userProfile.firstName,
+        email: userProfile.email,
+        isSchResident: userProfile.bmeStatus.includes(BmeUnitScope.BME_VIK_ACTIVE),
```
Comment (Isti01) [inline apps/backend/src/auth/auth.service.ts:23]: It think we need an isActiveVikStudent column in the enitity too

Comment (Tschonti) [reply apps/backend/src/auth/auth.service.ts]: do we even need these fields in the db? They will only ever be used when a registration period has ended, but then we will need to query authSch for up-to-date info, I don't see much use of storing this data.

**Review by Tschonti — APPROVED** [review]

**Pull request merged**